src/js/components/Select/Select.js

```diff
@@ -284,6 +286,11 @@ const Select = forwardRef(
 
     const iconColor = getIconColor(theme);
 
+    let ariaLabelledBy;
+    if (formFieldData?.inForm && id && !ariaLabel && !placeholder) {
+      ariaLabelledBy = `grommet-${id}__input__label ${id}`;
```

Comment:
If someone has a custom Select, our FormField code won't add the `__input` so I think we should make sure this stays in alignment with the `htmlFor`.

For example, the htmlFor on this won't be updated by our internal logic, so we'd be creating a broken aria-labelledby:

```
<FormField label="Label" htmlFor="myselect">
   <Select id="myselect" options={[...]} />
</FormField>
```